### DISCUSSION TITLE
Enums affect analysis

### DISCUSSION THREAD

**Discussion by MrQubo** [Vector35/binaryninja-api#6250] [Analysis] [answered]
My only use case for enums is to affect display of integers, so I can see, e.g. `AF_INET` instead of `2`. The problem is that sometimes changing the type of a struct field to an enum will break decompilation in unexpected ways, like instead of struct member access I get pointer arithmetic. That happened multiple times already, but I can't share binary, and cannot reproduce the issue.

I think, for the purposes of analysis enums should be treated exactly like their underlying type, and enums should only be taken into consideration when displaying integer values. Am I missing some beneficial use cases for enums in analysis?

**Comment by psifertex** [comment, marked as answer]
Can you give a more specific example of how they break? It's extremely difficult to reply to this in a meaningful way without more understanding of what is breaking for you and how. Enums are a type, changing the type could mean that you for example, replace a correctly sized type with an incorrectly sized one which would change the decompilation. 

That may be why you're not able to reproduce the issue.

Without more information though there's not really a whole lot we can do here unfortunately.
